lib/transaction.js

```diff
@@ -119,9 +123,9 @@ export class TransactionMap {
     }
 
     extend(infos: Collection<TransactionInfo>): TransactionMap {
-        let copy = new Map(this.infos);
+        let copy = this.infos;
         infos.forEach((info) => {
-            copy.set(info.id, info);
+            copy = copy.set(info.id, info);
```

Comment:
Immutable.js has probably some `extend` concept already. Actually, I think we can get rid of the whole TransactionMap now.